src/main/java/org/tarantool/jdbc/SQLConnection.java

```diff
@@ -171,25 +177,38 @@ public String nativeSQL(String sql) throws SQLException {
 
     @Override
     public void setAutoCommit(boolean autoCommit) throws SQLException {
+        checkNotClosed();
         if (autoCommit == false) {
             throw new SQLFeatureNotSupportedException();
         }
     }
 
     @Override
     public boolean getAutoCommit() throws SQLException {
+        checkNotClosed();
         return true;
     }
 
     @Override
     public void commit() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        checkNotClosed();
+        if (getAutoCommit()) {
+            throw new SQLNonTransientException(
+                    "Cannot commit when auto-commit is enabled.",
+                    SQLStates.IVALID_TRANSACTION_STATE.getSqlState()
+            );
+        }
```

Comment:
Please, add raising SQLFeatureNotSupportedException() or just TODO / XXX mark. Pass though here looks confusing for me. The same for rollback().